lib/GenomeFileUtil/core/GenomeInterface.py

```diff
@@ -429,16 +429,33 @@ def sizeof_fmt(num):
         if g['taxon_ref'] == "ReferenceTaxons/unknown_taxon":
             warnings.append('Unable to determine organism taxonomy')
 
-        print("Subobject Sizes:")
-        for x in ('cdss', 'mrnas', 'features', 'non_coding_features', 'ontologies_present'):
+        # MAX_GENOME_SIZE = 300000000 # UNCOMMENT TO TEST FAILURE MODE. Set to size needed
+        feature_lists = ('mrnas', 'features', 'non_coding_features','cdss')
+        master_key_sizes = dict()
+        need_to_remove_dna_sequence = _get_size(g) > MAX_GENOME_SIZE
+        for x in feature_lists:
             if x in g:
+                feature_type_dict_keys = dict()
+                for feature in g[x]:
+                    for feature_key in list(feature.keys()):
+                        if feature_key == "dna_sequence" and need_to_remove_dna_sequence:
```

Comment:
So I think the intent here is to check if the the genome is still too big before removing each dna sequences. (This code does not do this by the way because `need_to_remove_dna_sequence` is only calculated once) However, I don't think that's the best idea because it's a little more predictable for an genome to either have the sequence or not. I think we should just strip the DNA out of all features if the genome is too big but even if we don't, I think the granularity should be no smaller than the feature list.

Reply:
You can hit that level of granularity BTW by changing 437 to: `if x in g and _get_size(g) > MAX_GENOME_SIZE:` and removing `need_to_remove_dna_sequence`

Reply:
James I think you are misinterpreting the code.  It does this at the list level. As my print outs in slack show.

Reply:
Tell you what, you need add a test for this anyway. You can make a genome only removing the DNA sequence in mrna and features would be necessary but I will bet you a 🍺 that this code removes DNA sequence from all of the lists.

Reply:
James I did that in slack print outs